src/app/core/notifications/notifications_manager.nim

```diff
@@ -0,0 +1,183 @@
+    chatId: string, messageId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.NewMessage, sectionId: sectionId, 
+    chatId: chatId, messageId: messageId)
+    self.processNotification(title, message, details)
+
+  proc onShowMentionMessageNotification(self: NotificationsManager, title: string, message: string, sectionId: string, 
+    chatId: string, messageId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.NewMention, sectionId: sectionId, 
+    chatId: chatId, messageId: messageId)
+    self.processNotification(title, message, details)
+
+  proc onShowNewContactRequestNotification*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.NewContactRequest, sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onNewCommunityMembershipRequestNotification*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.JoinCommunityRequest, sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onMyRequestToJoinCommunityHasBeenAcccepted*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.MyRequestToJoinCommunityAccepted, 
+    sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc onMyRequestToJoinCommunityHasBeenRejected*(self: NotificationsManager, title: string, message: string, 
+    sectionId: string) {.slot.} =
+    let details = NotificationDetails(notificationType: NotificationType.MyRequestToJoinCommunityRejected, 
+    sectionId: sectionId)
+    self.processNotification(title, message, details)
+
+  proc processNotification(self: NotificationsManager, title: string, message: string, details: NotificationDetails) =
+    ## This is the main method which need to be called to process an event according to the preferences set in the 
+    ## notifications panel of the settings section.
+    ## 
+    ## This method determines whether a notification need to be displayed or not, what notification to display, whether
+    ## to display App or OS notification, what level of anonymous to apply, whether to play sound or not and so...
+
+    # I am not 100% sure about this.
+    # According to this:
+    # https://github.com/status-im/status-desktop/pull/4789#discussion_r805028513
+    # The app should use OS notification only in case it is not the active app or it's minimized at the moment 
+    # we're processing an event.
+    if(singletonInstance.localAccountSensitiveSettings.getUseOSNotifications() and 
+      app_isActive(singletonInstance.engine)):
```

Comment:
This condition can be moved to line 170, since we want to use OS notifications in the case the app is not at the front.
In other cases, we use local notifications (Toast).
The `useOSNotifications` is gonna be removed as well, since it was a legacy privacy feature that is no longer needed.

Reply:
@jrainville I added it deliberately here, cause that way we're avoiding a lot of unnecessary checks if we're gonna return from this proc anyway. So patter do it sooner than later. So I wouldn't change this if you agree.

> The useOSNotifications is gonna be removed as well, since it was a legacy privacy feature that is no longer needed.

Yes, I agree, but cannot prepare code now for something what is yet to come. So we will change that when the time comes.